source/server/admin/stats_handler.cc

```diff
@@ -140,7 +140,8 @@ Http::Code StatsHandler::handlerStats(absl::string_view url,
 Http::Code StatsHandler::handlerPrometheusStats(absl::string_view path_and_query,
                                                 Http::ResponseHeaderMap&,
                                                 Buffer::Instance& response, AdminStream&) {
-  const Http::Utility::QueryParams params = Http::Utility::parseQueryString(path_and_query);
+  const Http::Utility::QueryParams params =
+      Http::Utility::parseQueryString(path_and_query, /*do_percent_decoding=*/true);
```

Comment:
I think we should avoid the need for the comment to make clear what the true parameter means. Perhaps instead of a boolean flag we should use a separate method?

Reply:
Updated.